diff --git a/gsk/gpu/gskgpurenderpass.c b/gsk/gpu/gskgpurenderpass.c
index 41a0f99..d5d8da4 100644
--- a/gsk/gpu/gskgpurenderpass.c
+++ b/gsk/gpu/gskgpurenderpass.c
@@ -601,6 +601,7 @@ gsk_gpu_render_pass_push_transform (GskGpuRenderPass                 *self,
                 gsk_gpu_clip_init_empty (&self->clip, graphene_point_zero (), &new_bounds);
               }
           }
+        gsk_transform_unref (clip_transform);
 
         self->modelview = gsk_transform_scale (gsk_transform_ref (storage->modelview),
                                                self->scale.width,
